tests_tf/test_attacks.py

```diff
@@ -454,119 +454,6 @@ def test_attack_strength_np_batched(self):
     new_labs = np.argmax(self.sess.run(self.model.get_logits(x_adv)), axis=1)
     self.assertLess(np.mean(feed_labs == new_labs), 0.1)
 
```

Comment:
It's probably OK, but just so I know what's going on, can you explain why this was removed?

Reply:
I think all the methods here were moved to the TestMadryEtAl class. I don't fully understand why they needed to be moved. If that is what happened, I think it's probably not the right thing to do, but maybe I'm missing something. The MadryEtAl class doesn't do anything but initialize ProjectedGradientDescent with the right arguments, so it doesn't make a lot of sense for all the tests to be pushed into that highly specialized subclass.

Reply:
There were two `class TestBasicIterativeMethod`. I removed one and put the methods into the other.

Reply:
That's not actually how I'm reading the diff.
I think you moved all the tests into TestMadryEtAl.

I think what happened was there used to be a BasicIterativeMethod attack class. Then later we added the MadryEtAl attack class. Then later we implemented both using a common class called ProjectedGradientDescent, and made BasicIterativeMethod and MadryEtAl just be empty classes that instantiate ProjectedGradientDescent with the right arguments.

There are now TestBasicIterativeMethod and TestMadryEtAl which are basically just smoke tests to make sure those empty child classes can be called. But TestBasicIterativeMethod caused a problem because that was the same name as all the old tests.

I think the right way to fix it is just don't move anything, but change the name of the old TestBasicIterativeMethod to TestProjectedGradientDescent.